modules/annex/annex_archive.go

```diff
@@ -151,6 +186,20 @@ func createArchiveZip(ctx context.Context, repo *git.Repository, target io.Write
 				header.Name = oldHeader.Name
 				header.Method = zip.Deflate
 
+				if te.IsLink() {
+					// in the case of a symlink, git-annex stores the mode on the annex/objects/ file
+					// also, since *git* only defines 3 modes, symlink, regular, and executable
+					// (plus some internal modes; see git.EntryMode for the full list)
+					// to avoid getting confused by umask, just read the *owner*'s permission
+					if (stat.Mode().Perm() & 0o100) != 0 {
+						header.SetMode(fs.FileMode(git.EntryModeExec))
```

Comment:
Is this direct mapping correct? According to https://pkg.go.dev/code.gitea.io/gitea/modules/git#EntryMode git.EntryModeExec is `0o100755`. According to https://pkg.go.dev/io/fs#FileMode ModePerm is `0o777` and according to https://pkg.go.dev/io/fs#FileMode.IsRegular regular files should have no ModeType bits set. Shouldn't therefore git.EntryModeExec map to `fs.FileMode(0o755)` instead? This applies for targz and git.EntryModeBlob as well.